examples/c/5_series/gnss_ins/5_series_gnss_ins_example.c

```diff
@@ -1,354 +1,1106 @@
 
-    uint16_t filter_base_rate;
+    return 0;
+}
 
-    if(mip_3dm_filter_get_base_rate(&device, &filter_base_rate) != MIP_ACK_OK)
-         exit_gracefully("ERROR: Could not get filter base rate format!");
+////////////////////////////////////////////////////////////////////////////////
+/// @brief Custom logging callback for MIP SDK message formatting and output
+///
+/// @details Processes and formats log messages from the MIP SDK based on
+///          severity level. Routes messages to appropriate output streams -
+///          errors and fatal messages go to stderr while other levels go to
+///          stdout. Each message is prefixed with its severity level name.
+///
+/// @param _user Pointer to user data (unused in this implementation)
+/// @param _level Log message severity level from microstrain_log_level enum
+/// @param _format Printf-style format string for the message
+/// @param _args Variable argument list containing message parameters
+///
+void log_callback(void* _user, const microstrain_log_level _level, const char* _format, va_list _args)
+{
+    // Unused parameter
+    (void)_user;
 
-    const uint16_t filter_sample_rate = 100; // Hz
-    const uint16_t filter_decimation = filter_base_rate / filter_sample_rate;
+    switch (_level)
+    {
+        case MICROSTRAIN_LOG_LEVEL_FATAL:
+        case MICROSTRAIN_LOG_LEVEL_ERROR:
+        {
+            fprintf(stderr, "%s: ", microstrain_logging_level_name(_level));
+            vfprintf(stderr, _format, _args);
+            break;
+        }
+        case MICROSTRAIN_LOG_LEVEL_WARN:
+        case MICROSTRAIN_LOG_LEVEL_INFO:
+        case MICROSTRAIN_LOG_LEVEL_DEBUG:
+        case MICROSTRAIN_LOG_LEVEL_TRACE:
+        {
+            fprintf(stdout, "%s: ", microstrain_logging_level_name(_level));
+            vfprintf(stdout, _format, _args);
+            break;
+        }
+        case MICROSTRAIN_LOG_LEVEL_OFF:
+        default:
+        {
+            break;
+        }
+    }
+}
 
-    const mip_descriptor_rate filter_descriptors[5] = {
-        { MIP_DATA_DESC_FILTER_FILTER_TIMESTAMP, filter_decimation },
-        { MIP_DATA_DESC_FILTER_FILTER_STATUS,    filter_decimation },
-        { MIP_DATA_DESC_FILTER_POS_LLH,          filter_decimation },
-        { MIP_DATA_DESC_FILTER_VEL_NED,          filter_decimation },
-        { MIP_DATA_DESC_FILTER_ATT_EULER_ANGLES, filter_decimation },
+////////////////////////////////////////////////////////////////////////////////
+/// @brief Captures and configures device gyro bias
+///
+/// @param _device Pointer to the initialized MIP device interface
+///
+void capture_gyro_bias(mip_interface* _device)
+{
+    // Get the command queue so we can increase the reply timeout during the capture duration,
+    // then reset it afterward
+    mip_cmd_queue*    cmd_queue        = mip_interface_cmd_queue(_device);
+    const mip_timeout previous_timeout = mip_cmd_queue_base_reply_timeout(cmd_queue);
```

Comment:
We should probably have a better way to set/get the timeout directly, but that's not part of this PR.